cpp/include/cudf/types.hpp

```diff
@@ -234,6 +234,12 @@ enum class type_id : int32_t {
   NUM_TYPE_IDS  ///< Total number of type ids
 };
 
+/// @brief Indicates whether a function is null-aware or not.
+enum class null_aware : uint8_t {
```

Comment:
```suggestion
enum class null_aware : bool {
```
nit: could we just use `bool`?